### PR TITLE
[18.0] [IMP] web_m2x_options: Migration to 18.0

### PR DESCRIPTION
**Pull request by chaule97**
- I update the JavaScript code to follow the ECMAScript

**OCA - port**
- https://github.com/chaule97/web/pull/2

### CODE REVIEW

**Comment by xaviedoanhduy** [comment]
Hello @chaule97 Are you confused about the name of the PR? 
It should be [18.0][MIG] or [18.0] [IMP]?

File: web_m2x_options/static/src/components/base.xml [hunk 1]
```diff
@@ -0,0 +1,30 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+<!-- Copyright 2017 Jairo Llopis <jairo.llopis@tecnativa.com>
+     License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl). -->
+<templates xml:space="preserve">
+
+    <t
+        t-name="web_m2x_options.AutoComplete"
+        t-inherit="web.AutoComplete"
+        t-inherit-mode="extension"
+        owl="1"
```
Comment (xaviedoanhduy) [inline web_m2x_options/static/src/components/base.xml:10]:
```suggestion
```

> Remove owl="1" from OWL templates, as it's not needed anymore as all of them are OWL now. Reference: https://github.com/odoo/odoo/pull/130467. Application of this removal in Odoo core: https://github.com/odoo/odoo/pull/141383.

Comment (xaviedoanhduy) [inline web_m2x_options/static/src/components/base.xml]:
```suggestion
```

> Remove owl="1" from OWL templates, as it's not needed anymore as all of them are OWL now. Reference: https://github.com/odoo/odoo/pull/130467. Application of this removal in Odoo core: https://github.com/odoo/odoo/pull/141383.

File: web_m2x_options/static/src/components/base.xml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+        <xpath expr="//t[@t-foreach='source.options']/li/a" position="attributes">
+            <attribute name="t-attf-style">{{ option.style }}</attribute>
+        </xpath>
+    </t>
+
+     <t
+        t-name="web_m2x_options.Many2ManyTagsField"
+        t-inherit="web.Many2ManyTagsField"
+        t-inherit-mode="extension"
+        owl="1"
```
Comment (xaviedoanhduy) [inline web_m2x_options/static/src/components/base.xml]:
```suggestion
```
the same above

File: web_m2x_options/static/src/components/form.esm.js [hunk 3]
```diff
@@ -0,0 +1,327 @@
+/** @odoo-module **/
```
Comment (xaviedoanhduy) [inline web_m2x_options/static/src/components/form.esm.js]:
```suggestion
```
no need for /** @odoo-module */ anymore (`https://github.com/odoo/odoo/pull/142858`)

File: web_m2x_options/static/src/components/form.esm.js [hunk 4]
```diff
@@ -0,0 +1,327 @@
...
+    /**
+     * @override
+     * add more method to add subview limit on formview
+     */
+    async _setSubViewLimit() {
+        const ir_options = session.web_m2x_options || {};
+        const {fieldNodes: activeFields} = this.archInfo;
+        const isSmall = this.user;
+
+        const limit = parseInt(ir_options["web_m2x_options.field_limit_entries"], 10);
```
Comment (xaviedoanhduy) [inline web_m2x_options/static/src/components/form.esm.js:304]: very easy to broken! if the value in the `parseInt` does not return a numeric character.

File: web_m2x_options/static/src/components/form.esm.js [hunk 5]
```diff
@@ -0,0 +1,327 @@
...
+        return options;
+    },
+});
+
+patch(FormController.prototype, {
+    /**
+     * @override
+     */
+    setup() {
+        super.setup(...arguments);
```
Comment (xaviedoanhduy) [inline web_m2x_options/static/src/components/form.esm.js]:
```suggestion
        super.setup();
```

**Review by xaviedoanhduy — CHANGES_REQUESTED** [review]

**Comment by xaviedoanhduy** [comment]
btw, could you squash commits? 
https://github.com/OCA/maintainer-tools/wiki/Merge-commits-in-pull-requests

**Comment by xaviedoanhduy** [comment]
I get this error in console log when install module `website`:
```
Uncaught Error: Error while loading "@mail/views/web/fields/many2many_tags_email/many2many_tags_email":
Error: Invalid object: unknown key 'm2mOptionsPropsCreate', unknown key 'm2mOptionsPropsCreateEdit', unknown key 'm2mOptionsPropsLimit', unknown key 'm2mOptionsPropsSearchMore', unknown key 'm2mOptionsProps'
    at ModuleLoader.startModule (web.assets_web.min.js:17:225)
    at ModuleLoader.startModules (web.assets_web.min.js:16:57)
    at ModuleLoader.addJob (web.assets_web.min.js:13:39)
    at ModuleLoader.define (web.assets_web.min.js:12:127)
    at web.assets_web.min.js:16304:6
```
image

**Comment by chaule97** [comment]
> I get this error in console log when install module `website`:
> 
> ```
> Uncaught Error: Error while loading "@mail/views/web/fields/many2many_tags_email/many2many_tags_email":
> Error: Invalid object: unknown key 'm2mOptionsPropsCreate', unknown key 'm2mOptionsPropsCreateEdit', unknown key 'm2mOptionsPropsLimit', unknown key 'm2mOptionsPropsSearchMore', unknown key 'm2mOptionsProps'
>     at ModuleLoader.startModule (web.assets_web.min.js:17:225)
>     at ModuleLoader.startModules (web.assets_web.min.js:16:57)
>     at ModuleLoader.addJob (web.assets_web.min.js:13:39)
>     at ModuleLoader.define (web.assets_web.min.js:12:127)
>     at web.assets_web.min.js:16304:6
> ```
> 
> image

you can read [it](https://github.com/OCA/web/pull/2921), this issue still has no method

File: web_m2x_options/static/src/components/form.esm.js (hunk 4 above)
Comment (chaule97) [reply web_m2x_options/static/src/components/form.esm.js:304]: 10 is default if ir_options["web_m2x_options.field_limit_entries"] can't get number

**Pull request closed without merging**